gcovr/tests/test_gcovr.py

```diff
@@ -143,12 +157,23 @@ def run(cmd, cwd=None):
     return returncode == 0
 
 
-def find_reference_files(reference_dir, output_pattern):
-    assert os.path.isdir(reference_dir), "Reference directory exists"
-    for pattern in output_pattern:
-        for reference in glob.glob(os.path.join(reference_dir, pattern)):
-            coverage = os.path.basename(reference)
-            yield coverage, reference
+def find_reference_files(reference_dirs, output_pattern):
+    seen_files = set([])
+    for reference_dir in reference_dirs:
+        for pattern in output_pattern:
+            for dir in REFERENCE_DIRS:
+                if (platform.system() == "Windows") and os.path.isdir(reference_dir + "-Windows"):  # pragma: no cover
+                    for file in glob.glob(os.path.join(f"{dir}-Windows", pattern)):
+                        if os.path.basename(file) not in seen_files:
+                            coverage = os.path.basename(file)
+                            seen_files.add(coverage)
+                            yield coverage, file
```

Comment:
Code duplication between the `if` and `elif` branches. This duplication can likely be avoided by appending `-Windows` if appropriate, similar to the previous code:

```python
if platform.system() == "Windows" and os.path.isdir(f"{reference_dir}-Windows"):
    reference_dir += "-Windows"
```

Also, there is no need to mark Windows-only branches as `pragma: no cover` as we do collect coverage on Windows.

Reply:
The windows specific reference files are not present anymore therefore I will remove the special reference handling for windows.

Reply:
Forget my previous comment, the test `exclude-throw-branches` uses a seperate reference folder for Windows.

Reply:
On Windows both folders need to be checked. I added a inner loop to remove the code duplication.